### PR TITLE
#1049 - Fix serialization bug with EhcacheTicketRegistry

### LINKED ISSUE #1049: Cannot serialize ticket with EhcacheTicketRegistry
See https://issues.jasig.org/browse/CAS-1494

### PR DESCRIPTION
**Pull request by Ammar0**
Closes #1049

### CODE REVIEW

File: cas-server-core/src/main/java/org/jasig/cas/authentication/SuccessfulHandlerMetaDataPopulator.java [hunk 1]
```diff
@@ -36,7 +37,12 @@
 
     @Override
     public void populateAttributes(final AuthenticationBuilder builder, final Credential credential) {
-        builder.addAttribute(SUCCESSFUL_AUTHENTICATION_HANDLERS, new HashSet(builder.getSuccesses().keySet()));
+        Set<String> successes = builder.getSuccesses().keySet();
```
Comment (leleuj) [inline cas-server-core/src/main/java/org/jasig/cas/authentication/SuccessfulHandlerMetaDataPopulator.java]: For my understanding, when are the `successes` null? Should it ever happen?

Comment (mmoayyed) [inline cas-server-core/src/main/java/org/jasig/cas/authentication/SuccessfulHandlerMetaDataPopulator.java]: No it never should. When the populator runs, it is assumed that authN is successful. Need to double check, but I think that is the current behavior.

**Comment by mmoayyed** [comment]
@Ammar0 any updates on this?

**Comment by mmoayyed** [comment]
@leleuj in light of the next patch release, how about we merge this for now and fix up other issues locally? Don't think we can update this PR in time.

**Comment by mmoayyed** [comment]
This should also be merged into 4.0.x

**Comment by leleuj** [comment]
I'm not against it. At worst, it's a useless check.

**Pull request merged**